packages/react-composites/src/composites/common/AvatarPersona.tsx

```diff
@@ -67,11 +66,17 @@ export interface AvatarPersonaProps extends IPersonaProps {
 export const AvatarPersona = (props: AvatarPersonaProps): JSX.Element => {
   const { userId, dataProvider, text, imageUrl, imageInitials, initialsColor, initialsTextColor, showOverflowTooltip } =
     props;
-  const userIds = useMemo(() => {
-    return [userId];
-  }, [userId]);
 
-  const [data] = useCustomAvatarPersonaData(userIds, dataProvider);
+  const [data, setData] = useState<AvatarPersonaData | undefined>();
+
+  useEffect(() => {
+    (async () => {
+      if (dataProvider && userId) {
+        const data = await dataProvider(userId);
+        setData(data);
```

Comment:
Would be nice to do a deep compare of data object, if every properties in the data object is the same, then skip setData() function